Scripts/Python/xAgeSDLBoolActivatorComboSet.py

```diff
@@ -224,3 +247,22 @@ def _ValidateCombination(self):
         if ',' not in combination.value:
             PtDebugPrint("xAgeSDLBoolActComboSet._ValidateCombination():\tCombination does not have separation delimiters")
         self._combination = [int(i) for i in combination.value.split(',')]
+
+    def _CheckSolved(self):
+        if self._numCorrect == len(self._combination):
+            PtDebugPrint("xAgeSDLBoolActComboSet._CheckSolved():\tWoohoo! You have solved the puzzle!", level=kWarningLevel)
+            self._solved = True
+            if disableOnSolve.value:
+                self._butsInUse = True
+
+    def _AddToAttempt(self, actId):
+        attempt = self._attemptCombo
+        attempt.append(actId)
+        if len(attempt) > len(self._combination):
+            attempt = attempt[1:]
+        PtDebugPrint(f"xAgeSDLBoolActComboSet._AddToAttempt():\t New attempt value: {attempt}", level=kWarningLevel)
+        self._attemptCombo = attempt
+
+    def _IsAttemptCorrectAtIndex(self, i):
+        attempt = self._attemptCombo
+        return attempt[i:] == self._combination[0:len(attempt)-i]
```

Comment:
```suggestion
        return attempt[i:] == self._combination[:-i]
```

Reply:
I don't think that's the same thing, is it? I think I should be able to omit the 0 as the start of the range, but the end of the range is `len(attempt)-i`, not `len(self._combination)-i`, which I think is what your suggestion would be doing, right?

Reply:
Ah, sorry, I missed that we were indexing from the end of a different list.